lib/server.js

```diff
@@ -919,7 +920,9 @@ Server.prototype._run = function _run(req, res, route, chain, cb) {
                 // flag is true, we will avoid automatically calling cb at the
                 // end of this function, which causes server to move on to the
                 // next handler in the chain.
-                emittedError = self._emitErrorEvents(req, res, route, arg,
+                emittingErrors = true;
```

Comment:
There's a race condition here due to the way the code is structured. Basically, restify's internal cb is called in two places. 

When you `next(err)`, assuming you have error listeners attached, we will async fire the events serially, flush response, then call cb (L923-L925). 

If no listeners are attached, then `_emitErrorEvents` will return false, and cb is fired down at L1030. After `_emitErrorEvents` completes and calls its callback on L923, it will then call cb again at L928. Calling cb twice will throw, as intended. 

We are using the `emittingErrors` closure var to basically tell one pathway to not call cb since it will be called later on. This is not great and really hard to grok, but not something to solve in this PR.